tracing/scope.go

```diff
@@ -0,0 +1,59 @@
+package tracing
+
+import (
+	"context"
+	"runtime"
+	"strconv"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// Scope state.
+type S struct {
+	Ctx context.Context
```

Comment:
The use case of `Scope()` is to make it as simple as possible.  I want to send the action function a single parameter with minimal typing.  If later down the road I want to pass another piece of data, I don't have to make a breaking change by changing the function signature.

Reply:
I see. OK.

Reply:
I found a way to eliminate the `S` struct entirely.  The context currently contains the active span.  It might as well also hold the `Tracer` object.  That way, the action function only needs a context parameter and all Go devs will be comfortable with understanding that.

Reply:
Awesome.